src/media-segment-request.js

```diff
@@ -350,38 +364,58 @@ const handleSegmentBytes = ({
 }) => {
   const bytesAsUint8Array = new Uint8Array(bytes);
 
-  segment.isFmp4 =
-    // only set the property the first time we see some bytes so that partial appends
-    // don't try to check every section of bytes (since the check should only consider the
-    // first bytes in the segment)
-    typeof segment.isFmp4 === 'boolean' ? segment.isFmp4 : isLikelyFmp4Data(bytesAsUint8Array);
+  if (isLikelyFmp4Data(bytesAsUint8Array)) {
+    segment.isFmp4 = true;
+    const {tracks} = segment.map;
+
+    const trackInfo = {
+      hasVideo: !!tracks.video,
+      hasAudio: !!tracks.audio
+    };
+
+    if (tracks.audio && tracks.audio.codec) {
+      trackInfo.audioCodec = tracks.audio.codec;
+    }
+
+    if (tracks.video && tracks.video.codec) {
+      trackInfo.videoCodec = tracks.video.codec;
+    }
 
-  if (segment.isFmp4) {
+    // if an init segment has both audio and video is muxed
+    if (tracks.video && tracks.audio) {
+      trackInfo.hasAudio = false;
+      if (trackInfo.audioCodec) {
+        trackInfo.videoCodec += `,${trackInfo.audioCodec}`;
+        delete trackInfo.audioCodec;
+      }
+    }
 
     // since we don't support appending fmp4 data on progress, we know we have the full
     // segment here
-    const startTime = probeMp4StartTime(bytesAsUint8Array, segment.map.bytes);
-
-    // we don't parse fmp4 (yet), so we can't provide any track info, however, the
-    // track info callback should still be called before the dataFn is called
-    trackInfoFn(segment, null);
-    // use null for the media type since we don't technically know whether it's audio or
-    // video
+    trackInfoFn(segment, trackInfo);
     // the probe doesn't provide the segment end time, so only callback with the start
     // (the end time can be roughly calculated by the receiver using the duration)
-    timingInfoFn(segment, null, 'start', startTime);
-    dataFn(segment, { data: bytes });
+    const timingInfo = mp4probe.startTime(segment.map.timescales, bytesAsUint8Array);
+
+    if (trackInfo.hasAudio) {
+      timingInfoFn(segment, 'audio', 'start', timingInfo);
+    }
+
+    if (trackInfo.hasVideo) {
+      timingInfoFn(segment, 'video', 'start', timingInfo);
+    }
+    dataFn(segment, {data: bytes, type: trackInfo.hasAudio ? 'audio' : 'video'});
```

Comment:
I guess we assume fmp4 is always demuxed and that's why this works. A comment here would be helpful!

Reply:
If `track.hasAudio` is `true` at this point it is audio only, since if we have both audio and video we set `hasAudio` to `false` and merge the `audioCodec` into the `videoCodec`. I will add a comment though.

Reply:
Yeah, I meant that if you have a ts segment, you could have a muxed stream and alt audio in the one segment. But, since we're dealing with fmp4 here we assume it's demuxed so if `hasAudio` is true, it must be alt audio